### PR TITLE
Infiltrator MODs don't have a plasma visor anymore

### PR DESCRIPTION
**Pull request by Exester509**
## About The Pull Request

Disables the plasma stabilizer's visor for any MODsuit that has an infiltrator module

## Why It's Good For The Game

It doesn't make sense that a suit designed for concealing your identity gave away your species. This changes that, allowing plasmamen to freely use this MODsuit.

## Changelog

:cl: StaringGasMask
qol: Now plasmamen can use the infiltrator MODsuit without having their species revealed. The helmet's still not sealed, so remember your mask.
/:cl:

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
i feel like infiltrator module should have some sort of trait that gets added to the modsuit, but this implementation is fine enough tbh

**Review by Fikou — CHANGES_REQUESTED** [review]
The way this is coded isn't good but in general I don't see the purpose of this limitation. The fact you're a plasmaman doesn't reveal your full identity, this doesn't really free you from anything. You also don't need the module, you could just be under a shower while putting the suit on.

**Comment by Helg2** [comment]
> The fact you're a plasmaman doesn't reveal your full identity

Considering that there's 5% of plasmeme at max for crew, and almost every job has its unique helmet paints, it does in-fact reveals your identity, if sec is smart enough of course but still.

**Comment by Fikou** [comment]
> > The fact you're a plasmaman doesn't reveal your full identity
> 
> Considering that there's 5% of plasmeme at max for the crew, and almost every job has its unique helmet paints, it does in-fact reveal your identity, if sec is smart enough of course but still.

how do their helmet paints matter? they are completely unrelated in this scenario, you arent wearing them

**Comment by Helg2** [comment]
Im just being idiotic, nvm

**Comment by Exester509** [comment]
> The way this is coded isn't good but in general I don't see the purpose of this limitation. The fact you're a plasmaman doesn't reveal your full identity, this doesn't really free you from anything. You also don't need the module, you could just be under a shower while putting the suit on.

No, in this case you actually need the module, since the MOD isn't sealed, and doesn't prevent your ignition, even if that trick might work with other MODsuits
Plasmamen are forced to have the purple visor in the current state, and while it's true that it doesn't completely reveal your identity, it doesn't have any workaround, like other things like your backpack or TTS voice. It doesn't help that plasmamen are one of the least played species, resulting in a dead giveaway.

File: code/modules/mod/modules/modules_general.dm [hunk 1]
```diff
@@ -471,6 +471,13 @@
 	incompatible_modules = list(/obj/item/mod/module/plasma_stabilizer)
 	overlay_state_inactive = "module_plasma"
 
+/obj/item/mod/module/plasma_stabilizer/add_module_overlay()
+	if(locate(/obj/item/mod/module/infiltrator) in mod.modules)
+		overlay_state_inactive = null
+	else
+		overlay_state_inactive = initial(overlay_state_inactive)
+	..()
```
Comment (Fikou) [inline code/modules/mod/modules/modules_general.dm:479]: this is a bad way to do this.
i dont see why you would wanna set the overlay state here
just do an early return in generate_worn_overlay of this module

Comment (Exester509) [reply code/modules/mod/modules/modules_general.dm]: Done, sorry for that, I'm not very familiar with the code

File: code/modules/mod/modules/modules_general.dm [hunk 2]
```diff
@@ -471,6 +471,11 @@
 	incompatible_modules = list(/obj/item/mod/module/plasma_stabilizer)
 	overlay_state_inactive = "module_plasma"
 
+/obj/item/mod/module/plasma_stabilizer/generate_worn_overlay()
+	if(locate(/obj/item/mod/module/infiltrator) in mod.modules)
+		return
+	. = ..()
```
Comment (Fikou) [inline code/modules/mod/modules/modules_general.dm:477]:
```suggestion
	return ..()
```

File: code/modules/mod/modules/modules_general.dm [hunk 3]
```diff
@@ -471,6 +471,11 @@
 	incompatible_modules = list(/obj/item/mod/module/plasma_stabilizer)
 	overlay_state_inactive = "module_plasma"
 
+/obj/item/mod/module/plasma_stabilizer/generate_worn_overlay()
+	if(locate(/obj/item/mod/module/infiltrator) in mod.modules)
+		return
```
Comment (Fikou) [inline code/modules/mod/modules/modules_general.dm:476]:
```suggestion
		return list()
```

**Comment by GuillaumePrata** [comment]
Species should have pros and cons. neutering downsides to make being an antag play the same for multiple species makes species selection boring.

If security seens this small detail of the modsuit and go after you, kudos for them to being knowledgeable and using the investigative side of their job.

**Comment by Exester509** [comment]
> Species should have pros and cons. neutering downsides to make being an antag play the same for multiple species makes species selection boring.
> 
> If security seens this small detail of the modsuit and go after you, kudos for them to being knowledgeable and using the investigative side of their job.

But it's not a small detail, it's a very visible visor that goes against the base concept of the suit, that is, providing anonimity. In my eyes, this is equivalent to an esword dealing half the damage if you're a moth, for example. Plasmamen already need to add the module, sacrificing complexity in order to even be able to use it. That's the way I see it, at least

**Comment by Exester509** [comment]
@Fikou I'm closing this for the time being, I'll reopen it when you're ready for reviewing it

**Comment by Fikou** [comment]
is it tested? @Exester509

**Comment by Exester509** [comment]
> is it tested? @Exester509

Yes, I've playtested a bunch of times and it works fine on the local

**Pull request merged**